src/airflow/dags/airnow.py

```diff
@@ -0,0 +1,153 @@
+        from airqo_etl_utils.airnow_utils import AirnowDataUtils
+
+        airnow_api = AirNowApi()
+        networks = airnow_api.get_networks()
+        extracted_bam_data = pd.DataFrame()
+
+        for network in networks:
+            if network["data_source"] == DataSource.AIRNOW:
+                network_data = AirnowDataUtils.extract_bam_data(
+                    api_key=network["api_key"],
+                    start_date_time=start_date_time,
+                    end_date_time=end_date_time,
+                )
+                extracted_bam_data = pd.concat(
+                    [extracted_bam_data, network_data], ignore_index=True
+                )
+
+        return extracted_bam_data
+
+    @task()
+    def process_data(airnow_data: pd.DataFrame):
+        from airqo_etl_utils.airnow_utils import AirnowDataUtils
+
+        return AirnowDataUtils.process_bam_data(data=airnow_data)
+
+    @task()
+    def send_to_bigquery(data: pd.DataFrame):
+        from airqo_etl_utils.bigquery_api import BigQueryApi
+        from airqo_etl_utils.constants import Tenant
+        from airqo_etl_utils.data_validator import DataValidationUtils
+
+        big_query_api = BigQueryApi()
+        table = big_query_api.bam_measurements_table
+        data = DataValidationUtils.process_for_big_query(
+            dataframe=data, tenant=Tenant.US_EMBASSY, table=table
+        )
+        big_query_api.load_data(data, table=big_query_api.bam_measurements_table)
+
+    extracted_bam_data = extract_bam_data()
+    processed_bam_data = process_data(extracted_bam_data)
+    send_to_bigquery(processed_bam_data)
+
+
+@dag(
+    "Airnow-Realtime-Bam-Data",
+    schedule="30 * * * *",
+    default_args=AirflowUtils.dag_default_configs(),
+    catchup=False,
+    tags=["bam", "airnow", "realtime"],
+)
+def airnow_bam_realtime_data():
+    import pandas as pd
+
+    @task()
+    def extract_bam_data():
+        from airqo_etl_utils.airnow_utils import AirnowDataUtils
+        from airqo_etl_utils.date import DateUtils
+
+        start_date_time, end_date_time = DateUtils.get_query_date_time_values()
+
+        networks = AirNowApi().get_networks()
+        extracted_bam_data = pd.DataFrame()
+        for network in networks:
+            if network["data_source"] == DataSource.AIRNOW:
+                network_data = AirnowDataUtils.extract_bam_data(
+                    api_key=network["api_key"],
+                    start_date_time=start_date_time,
+                    end_date_time=end_date_time,
+                )
+                extracted_bam_data = pd.concat(
+                    [extracted_bam_data, network_data], ignore_index=True
+                )
+
+        return extracted_bam_data
+
+    @task()
+    def process_data(data: pd.DataFrame):
+        from airqo_etl_utils.airnow_utils import AirnowDataUtils
+
+        processed_data =pd.DataFrame()
+        networks = AirNowApi().get_networks()
+        for network in networks:
```

Comment:
This looks like a repetition. The first tasks already attached the network/tenant @BenjaminSsempala